androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/ui/Utils.kt

```diff
@@ -0,0 +1,11 @@
+import androidx.compose.ui.Modifier
+
+/**
+ * Applies the supplied [modifier] to the chain only if [condition] is `true`.
+ */
+fun Modifier.modifyIf(condition: Boolean, modifier: Modifier): Modifier =
+    if (condition) {
+        this.then(modifier)
+    } else {
+        this
+    }
```

Comment:
What happens if this is true? Never seen it before therefore i ask

Reply:
Since the function is an extension of `Modifier`, `this` refers to the modifier on which the function is called (the terminology is that `this` of type `Modifier` is the _receiver_ of the function `modifyIf`).
So in this case, if i have the call-site as:
```kotlin
Modifier
    .padding(start = 8.dp)
    .modifyIf(clickableText, Modifier.clickable { changeState() })
```
then `Modifier.padding(start = 8.dp)` would be represented as `this`, and if the `condition` `clickableText` is `true`, we would apply `.then()` with the new modifier on top of `this`.